public/app/features/variables/inspect/VariablesDependenciesButton.tsx

```diff
@@ -21,6 +21,10 @@ export const UnProvidedVariablesDependenciesButton: FC<Props> = ({ variables })
   const nodes = useMemo(() => createDependencyNodes(variables), [variables]);
   const edges = useMemo(() => createDependencyEdges(variables), [variables]);
 
+  if (!edges.length) {
+    return null;
+  }
+
```

Comment:
I noticed the dependency graph button is visible all the time, so this hides it if there are no dependencies. Not exactly part of this PR though.